packages/aws-cdk-lib/aws-ecs/lib/credential-spec.ts

```diff
@@ -0,0 +1,139 @@
+import { IBucket } from '../../aws-s3';
+import { IParameter } from '../../aws-ssm';
+
+/**
+ * Base construct for a credential specification (CredSpec).
+ */
+export class CredentialSpec {
+  /**
+   * Get the ARN for an S3 object.
+   */
+  protected static arnForS3Object(bucket: IBucket, key: string) {
+    if (!key) {
+      throw new Error('key is undefined');
+    }
+
+    return bucket.arnForObjects(key);
+  }
+
+  /**
+   * Get the ARN for a SSM parameter.
+   */
+  protected static arnForSsmParameter(parameter: IParameter) {
+    return parameter.parameterArn;
+  }
+
+  /**
+   * Prefix string based on the type of CredSpec.
+   */
+  public prefixId: string;
+
+  /**
+   * Location or ARN from where to retrieve the CredSpec file.
+   */
+  public fileLocation: string;
+
+  /**
+   * @param fileLocation Location or ARN from where to retrieve the CredSpec file
+   */
+  constructor(prefixId: string, fileLocation: string) {
+    this.prefixId = prefixId;
+    this.fileLocation = fileLocation;
+  }
+
+  /**
+   * Called when the container is initialized to allow this object to bind
+   * to the stack.
+   */
+  public bind(): CredentialSpecConfig {
+    return {
+      typePrefix: this.prefixId,
+      location: this.fileLocation,
+    };
+  }
+}
+
+/**
+ * Credential specification (CredSpec) file.
+ */
+export class DomainJoinedCredentialSpec extends CredentialSpec {
+  /**
+   * Prefix Id for this type of CredSpec.
+   */
+  public static readonly PREFIX_ID = 'credentialspec';
```

Comment:
do we need this as a public variable? don't think it needs to be

Reply:
done, moved to constructor